s3/src/main/scala/akka/stream/alpakka/s3/scaladsl/S3Client.scala

```diff
@@ -41,6 +44,104 @@ final case class ListBucketResultContents(
+   * </p>
+   *
+   * @return The hex encoded MD5 hash of the content for the associated object
+   *         as calculated by Amazon S3.
+   */
+  lazy val eTag: Option[String] = metadata.collectFirst {
+    case e: ETag => e.etag.value.drop(1).dropRight(1)
+  }
+
+  /**
+   * <p>
+   * Gets the Content-Length HTTP header indicating the size of the
+   * associated object in bytes.
+   * </p>
+   * <p>
+   * This field is required when uploading objects to S3, but the AWS S3 Java
+   * client will automatically set it when working directly with files. When
+   * uploading directly from a stream, set this field if
+   * possible. Otherwise the client must buffer the entire stream in
+   * order to calculate the content length before sending the data to
+   * Amazon S3.
+   * </p>
+   * <p>
+   * For more information on the Content-Length HTTP header, see <a
+   * href="http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.13">
+   * http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.13</a>
+   * </p>
+   *
+   * @return The Content-Length HTTP header indicating the size of the
+   *         associated object in bytes.
+   * @see ObjectMetadata#setContentLength(long)
+   */
+  lazy val contentLength: Long =
+    metadata
+      .collectFirst {
+        case cl: `Content-Length` => cl.length
+      }
+      .getOrElse(0)
+
+  /**
+   * <p>
+   * Gets the Content-Type HTTP header, which indicates the type of content
+   * stored in the associated object. The value of this header is a standard
+   * MIME type.
+   * </p>
+   * <p>
+   * When uploading files, the AWS S3 Java client will attempt to determine
+   * the correct content type if one hasn't been set yet. Users are
+   * responsible for ensuring a suitable content type is set when uploading
+   * streams. If no content type is provided and cannot be determined by
+   * the filename, the default content type, "application/octet-stream", will
+   * be used.
+   * </p>
+   * <p>
+   * For more information on the Content-Type header, see <a
+   * href="http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.17">
+   * http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.17</a>
+   * </p>
+   *
+   * @return The HTTP Content-Type header, indicating the type of content
+   *         stored in the associated S3 object.
+   * @see ObjectMetadata#setContentType(String)
+   */
+  lazy val contentType: Option[String] = metadata.collectFirst {
+    case ct: ContentType => ct.value
+  }
+
+  /**
+   * Gets the value of the Last-Modified header, indicating the date
+   * and time at which Amazon S3 last recorded a modification to the
+   * associated object.
+   *
+   * @return The date and time at which Amazon S3 last recorded a modification
+   *         to the associated object.
+   */
+  lazy val lastModified: DateTime = metadata.collectFirst {
+    case ct: `Last-Modified` => ct.date
+  }.get
+
+}
```

Comment:
Ah, here they are done in the right way, so I guess the other methods were just left overs?

Reply:
yep, also the related constants have been removed